segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -0,0 +1,204 @@
+    This function will return a random dist matrix such that the distance between cells
+    of types A and B are overall larger than the distance between cells of types A and C
+
+    Each row and column representing a cell.
+    We generate the points using Gaussian distributions
+    Ideally, the parameters for A to B distances will be set such that they produce a lower range of values
+    than A to C distances.
+
+    Will return a random distance matrix as well as the dictionary of associated cell: label IDs
+    The above is generated from the generate_labels function
+
+    Args:
+        num_A: the number of A cells we wish to generate. Default 100
+        num_B: the number of B cells we wish to generate. Default 100
+        num_C: the number of C cells we wish to generate. Default 100
+        distr_AB: if specified, will be a dict listing the mean and variance of the Gaussian distribution
+            we wish to generate numbers from. If None, use the default values.
+        distr_AC: similar to dist_AB. Default will have a higher mean value.
+        seed: whether to fix the random seed or not. Useful for testing.
+            Should be a specified integer value. Default None.
+    """
+
+    # set the mean and variance of the Gaussian distributions of both AB and AC distances
+    if not distr_AB:
+        mean_ab = AB_DIST_MEAN
+        var_ab = AB_DIST_VAR
+    else:
+        mean_ab = distr_AB['mean']
+        var_ab = distr_AB['var']
+
+    if not distr_AC:
+        mean_ac = AC_DIST_MEAN
+        var_ac = AC_DIST_VAR
+    else:
+        mean_ac = distr_AC['mean']
+        var_ac = distr_AC['var']
+
+    # set random seed if set
+    if seed:
+        np.random.seed(seed)
+
+    # we initialize the random distances across different types of points
+    # note that we don't really care about aa, bb, bc, or cc, so we
+    # initialize those to garbage. We do need them for a proper
+    # distance matrix format, however.
+    random_aa = np.random.normal(0, 1, (num_A, num_A))
+    random_ab = np.random.normal(mean_ab, var_ab, (num_A, num_B))
+    random_ac = np.random.normal(mean_ac, var_ac, (num_A, num_C))
+    random_bb = np.random.normal(0, 1, (num_B, num_B))
+    random_bc = np.random.normal(0, 1, (num_B, num_C))
+    random_cc = np.random.normal(0, 1, (num_C, num_C))
+
+    # create each row one-by-one first
+    # we need to correct each aa, bb, and cc matrix to ensure symmetry
+    first_row = np.concatenate(((random_aa + random_aa.T) / 2, random_ab, random_ac), axis=1)
+    second_row = np.concatenate((random_ab.T, (random_bb + random_bb.T) / 2, random_bc), axis=1)
+    third_row = np.concatenate((random_ac.T, random_bc.T, (random_cc + random_cc.T) / 2), axis=1)
+
+    # then concatenate them together
+    dist_mat = np.concatenate((first_row, second_row, third_row), axis=0)
+
+    # assert that the created submatrix is symmetric
+    assert np.allclose(dist_mat, dist_mat.T, rtol=1e-05, atol=1e-08)
+
+    return dist_mat
```

Comment:
This method of generating a random distance matrix can create non-physical results.  Consider the case of just 3 cell-A's in a 2d plane, with distances a12, a23, and a13.  There are two geometric cases for this:

1. The cells are co-linear, so some reordering of a12 + a23 = a13 must be true
2. The cells form a triangle, and hence follow the all of the inequalities:
      - a12 + a23 < a13
      - a13 + a23 < a12
      - a12 + a13 < a23

This method of generating distance matrices doesn't guarantee these criteria and creates un-physical situations.  Rather than fix this, it's probably best remove this function and use `get_random_centroid_centers` into `point_init_dist_matrix` instead, that defaults to having mean_A closer to mean_B than mean_C